src/coreHandler.ts

```diff
@@ -75,9 +75,11 @@ export class CoreHandler {
 		})
 		this.core.onDisconnected(() => {
 			this.logger.info('Core Disconnected!')
+			this.setStatus(P.StatusCode.WARNING_MAJOR, ['Core Disconnected'])
 		})
 		this.core.onError((err) => {
 			this.logger.error('Core Error: ' + (err.message || err.toString() || err))
+			this.setStatus(P.StatusCode.BAD, ['Core error'])
```

Comment:
Do we know which errors are received here?
E.g. will an arbitrary error in core cause the gateway to have a bad status?

Reply:
The one we have seen a few times now is
error: Core Error: ddpError: connect ECONNREFUSED 172.18.0.4:80

But if this indicates that the gateway can't connect to core, can core then connect to the gateway to get the status?

Reply:
The bug itself also sounds a bit like a race condition in core connection / ddp client. Maybe we had some calls queued just before losing the connection, that started to execute when it was re-established and interfered with the new ones

Reply:
It looks like it's stalling in onConnectionRestored. Maybe waiting for some promise to complete.
I could set the status to unknow when reconnecting, but I don't think it'll make any difference, just that the status will be unknown instead of bad. And it might confuse people that the status is reconnecting but it'll never move on from that

Reply:
Looking at the logs, the rundownWatcher stop writing in the log.
So the problem might be that the watcher loop breaks and is not restarted.
I have made a change so it now restarts the watcher when restoring the connection